process/block/metablock.go

```diff
@@ -430,6 +430,18 @@ func (mp *metaProcessor) checkProofsForShardData(header *block.MetaBlock) error
 		if !mp.proofsPool.HasProof(shardData.ShardID, shardData.HeaderHash) {
 			return fmt.Errorf("%w for header hash %s", process.ErrMissingHeaderProof, hex.EncodeToString(shardData.HeaderHash))
 		}
+
+		prevProof := shardData.GetPreviousProof()
+		headersPool := mp.dataPool.Headers()
+		prevHeader, err := headersPool.GetHeaderByHash(prevProof.GetHeaderHash())
```

Comment:
here we check if there is shardData.HeaderHash proof in pool, and then we check proof fields on previousProof, is this ok?

Reply:
yes, the proof from pool should have already been validated